### ISSUE TITLE
Modularize the Online Model

### ISSUE THREAD

**Issue by dreadmill** [openxal/openxal#11]
There are several aspects to this issue.  The online model must first be made autonomous. All references outside the xal.model.* package, other that the xal.tools.* package, must be eliminated.  In particular, all references to SMF must be eliminated. A well-defined interface must be created for the online model.  This action allows for its treatment as a plugin whereas any "online model" capable of exposing that interface could be used.  The interface is a major effort due to its requirements of a standard for future use; perhaps the Scenario object is a good place to start.  The Synchronization manager will likely need to be upgraded to a formal set of association classes between the SMF machine representation and the online model interface.  It is currently a text-based property map.

To recap:
1. Remove all outside references except for the xal.tools.* package.
2. Design and implement a universal interface (say "OMI") for the online model (say OLMI).
3. Implement the online model as a "plugin" or otherwise expose the online model interface OMI.
4. Upgrade the Synchronization Manager as an aggregation of association classes between SMF and the OMI.
